pandas/io/stata.py

```diff
@@ -1047,30 +1033,30 @@ def __init__(
 
         if isinstance(path_or_buf, (str, bytes)):
             self.path_or_buf = open(path_or_buf, "rb")
-        else:
+        elif isinstance(path_or_buf, IOBase):
```

Comment:
Are there any sources that this would hit which define `read` but don't inherit from IOBase? Somewhat wary of making a change like this

Reply:
I rolled this one back because I can't guarantee that this isn't a restriction over `hasattr("write")`.